### PR TITLE
ci: switch from testing against js-ipfs to helia

### LINKED ISSUE #10013: Test interoperability with Helia and remove dependency on ipfs/interop
## Done Criteria
1. Kubo no longer relies on https://github.com/ipfs/interop for interoperability testing
2. Kubo CI checks to ensure it doesn't break compatibility with Helia
3. https://github.com/ipfs/interop is archived

## Why Important
We don't have Helia interoperability testing.
Relying on js-ipfs testing is keeping things like mplex around even though it should be deprecated: https://github.com/ipfs/kubo/issues/9958

## Notes
https://github.com/ipfs/interop is testing js-ipfs/js-ipfs, js-ipfs/kubo, and kubo/kubo.
js-ipfs is deprecated: https://github.com/ipfs/js-ipfs/issues/4336
We're not covering Kubo/Helia in the Kubo CI.
Kubo tests cover the Kubo/Kubo case better. 

Ideally, we run https://github.com/ipfs/helia/tree/main/packages/interop as part of Kubo CI and configure it to test the latest blessed Helia release against any PR's against Kubo master/main.

### PR DESCRIPTION
**Pull request by Jorropo**
Fixes #10013

### CODE REVIEW

**Comment by hacdias** [comment]
Just a quick comment: removing the following should also be done (as well as any other instructions that relate to `interop`):

https://github.com/ipfs/kubo/blob/89b0432387f0acfd87cd094d78faf7cd03a2af63/docs/RELEASE_ISSUE_TEMPLATE.md?plain=1#L177-L182

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -48,45 +48,61 @@ jobs:
         with:
           name: kubo
           path: cmd/ipfs/ipfs
-  interop:
+  helias:
```
Comment (SgtPooki) [inline .github/workflows/build.yml:51]:
```suggestion
  Helia-interop:
```

**Comment by BigLep** [comment]
> Just a quick comment: removing the following should also be done (as well as any other instructions that relate to `interop`):
> cc @galargh to remove the command from `kuboreleaser`

Good call on updating the release process.
We (Kubo maintainters) should go and own updating kuboreleaser.  It's Kubo's tool (that IPDX built for us).  Obviously a nice bonus if IPDX can pick it up, but we shouldn't wait on them.

**Review by hacdias — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -48,45 +48,61 @@ jobs:
         with:
           name: kubo
           path: cmd/ipfs/ipfs
-  interop:
+  Helia-interop:
```
Comment (hacdias) [inline .github/workflows/build.yml:51]:
```suggestion
  helia-interop:
```

Personally don't like the capitalisation here, but I don't care much.

**Comment by galargh** [comment]
Post merge LGMT, thank you for doing this ❤️

**Pull request merged**